src/main/scala/se/nimsa/sbx/app/routing/TransactionRoutes.scala

```diff
@@ -83,10 +83,16 @@ trait TransactionRoutes {
             }
           } ~ pathPrefix("outgoing") {
             path("poll") {
-              get {
-                onSuccess(boxService.ask(PollOutgoing(box)).mapTo[Option[OutgoingTransactionImage]]) {
-                  case Some(outgoingTransactionImage) => complete(outgoingTransactionImage)
-                  case None => complete(NotFound)
+              parameter("n".as[Long].?(1L)) { n =>
+                get {
+                  onSuccess(boxService.ask(PollOutgoing(box, n)).mapTo[Seq[OutgoingTransactionImage]]) { transactionImages =>
+                    if (transactionImages.isEmpty)
+                      complete(NotFound)
+                    else if (transactionImages.lengthCompare(1) == 0)
+                      complete(transactionImages.head)
+                    else
+                      complete(transactionImages)
+                  }
```

Comment:
Extended the API such that a list of pending outgoing files can be requested, instead of just one at a time. Backwards compability is ensured.